src/classes/CLAN_PrimaryLanguage_TDTM.cls

```diff
@@ -0,0 +1,240 @@
+                    continue;
+                }
+
+                // The Contact Language is no longer the Primary Language. Store the Language ID in a map to clear from the Contact later.
+                languageByContact.put(oldContactLanguage.Contact__c, oldContactLanguage.Language__c);
+            }
+
+            // Clear the Primary Language lookup for all Contacts that no longer have a Primary Language.
+            if (languageByContact.size() > 0) {
+                clearPrimaryLanguageOnContacts(contactsToUpdate.keySet(), languageByContact);
+            }
+        }
+
+        // Add Contact records to dmlWrapper to be updated by the TDTM_TriggerHandler class
+        dmlWrapper.objectsToUpdate.addAll((List<SObject>)contactsToUpdate.values());
+
+        // Are there any Contact Language records to update? Update them now so that we prevent trigger recursion.
+        if (contactLanguagesToUpdate.size() > 0) {
+            update contactLanguagesToUpdate.values();
+        }
+
+        // We're done with our updates, unset the recursion flag.
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CLAN_PrimaryLanguage_TDTM, false);
+
+        return dmlWrapper;
+    }
+
+    /*******************************************************************************************************
+    * @description Queries for Contact Language records that are marked as the Primary Language for the given
+    * set of Contacts, but not identified in the given set of primary Contact Languages. These Contact Languages 
+    * are then added to the contactLanguagesToUpdate private map to be updated to no longer have the Primary 
+    * Language checkbox set.
+    * @param contacts The set of Contacts associated to the new Primary Contact Language records.
+    * @param primaryContactLanguages The set of new Primary Contact Language records.
+    ********************************************************************************************************/
+    private void clearOldPrimaryContactLanguages(Set<Id> contacts, Set<Id> primaryContactLanguages) {
+        if (contacts == null || primaryContactLanguages == null) { // Should never happen, but check it just in case...
+            return;
+        }
+
+        // Query for any existing Contact Language records for the Contacts with Primary Language set to true, but not
+        // in the given set of primary Contact Languages.
+        List<Contact_Language__c> oldPrimaryContactLanguages = 
+            [SELECT Id, Primary_Language__c FROM Contact_Language__c 
+                WHERE Id NOT IN :primaryContactLanguages AND Contact__c IN :contacts AND Primary_Language__c = true];
+        
+        // Update the Contact Language records to no longer be marked as the Primary Language.
+        for (Contact_Language__c oldPrimaryContactLanguage : oldPrimaryContactLanguages) {
+            oldPrimaryContactLanguage.Primary_Language__c = false;
```

Comment:
I don't really know if there is a use case where a customer may want to be able to have more than one primary language defined -  I wouldn't think so. But if they do, I'd expect they can just turn off this trigger.

Reply:
Correct. I suppose someone always could, but I would go so far as to say it's an edge case and we're not going to support it outside of letting them turn off the trigger.